fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/translation/CompositeTranslator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * CompositeTranslator.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades.values.translation;
+
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.query.plan.cascades.AliasMap;
+import com.apple.foundationdb.record.query.plan.cascades.TranslationMap;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.google.common.base.Verify;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * This class represents a functional composition of a number of {@link Translator}s.
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class CompositeTranslator extends Translator {
+
+    @Nonnull
+    private final Optional<TrivialTranslator> simpleTranslatorMaybe;
```

Comment:
should be `xxxOptional`. `XXXMaybe()` is for methods.

Reply:
Translators are removed.